persistence/src/commonMain/kotlin/com/quran/shared/persistence/repository/BookmarkRepository.kt

```diff
@@ -0,0 +1,79 @@
+package com.quran.shared.persistence.repository
+
+import com.quran.shared.persistence.model.Bookmark
+import kotlinx.coroutines.flow.Flow
+
+class DuplicateBookmarkException(message: String) : Exception(message)
+class BookmarkNotFoundException(message: String) : Exception(message)
+
+interface BookmarkRepository {
```

Comment:
I would go with a repository per data type (single responsibility and separate between the data types). In this case, we would have PageBookmarksRepsitory and AyahBookmarksRepository.